distributed/scheduler.py

```diff
@@ -2031,7 +2060,7 @@ def _transition(
             if ts.state == "forgotten" and tg.name in self.task_groups:
                 # Remove TaskGroup if all tasks are in the forgotten state
                 if all(v == 0 or k == "forgotten" for k, v in tg.states.items()):
-                    ts.prefix.groups.remove(tg)
+                    ts.prefix.remove_group(tg)
                     del self.task_groups[tg.name]
 
```

Comment:
I wonder if this cannot be encapsulated in the `TaskState.transition` method

Reply:
Well, I guess not entirely. The `del self.task_groups[tg.name]` would still have to be done here. never mind then